miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt

```diff
@@ -165,45 +175,31 @@ abstract class MiniAppMessageBridge {
 
     @Suppress("LongMethod")
     private fun onRequestCustomPermissions(jsonStr: String) {
-        var callbackObj: CustomPermissionCallbackObj? = null
+        // initialize required properties using JsonStr before executing operations
+        customPermissionBridgeDispatcher.initCallBackObject(jsonStr)
 
-        try {
-            callbackObj = Gson().fromJson(jsonStr, CustomPermissionCallbackObj::class.java)
-            val permissionObjList = arrayListOf<CustomPermissionObj>()
-            callbackObj.param?.permissions?.forEach {
-                permissionObjList.add(CustomPermissionObj(it.name, it.description))
-            }
-
-            val permissionsWithDescription = MiniAppCustomPermissionManager()
-                .preparePermissionsWithDescription(permissionObjList)
+        // check if there is any denied permission
+        val deniedPermissions = customPermissionBridgeDispatcher.filterDeniedPermissions()
 
-            requestCustomPermissions(
-                permissionsWithDescription
-            ) { permissionsWithResult ->
-                // store values in SDK cache
-                val miniAppCustomPermission = MiniAppCustomPermission(
-                    miniAppId = miniAppInfo.id,
-                    pairValues = permissionsWithResult
-                )
-                customPermissionCache.storePermissions(miniAppCustomPermission)
-
-                // send JSON response to miniapp
-                onRequestCustomPermissionsResult(
-                    callbackId = callbackObj.id,
-                    jsonResult = MiniAppCustomPermissionManager().createJsonResponse(
-                        customPermissionCache,
-                        miniAppCustomPermission.miniAppId,
-                        permissionsWithDescription
+        if (deniedPermissions.isNotEmpty()) {
+            try {
+                requestCustomPermissions(
+                    deniedPermissions
+                ) { permissionsWithResult ->
+                    customPermissionBridgeDispatcher.sendHostAppCustomPermissions(
+                        permissionsWithResult
                     )
-                )
-            }
-        } catch (e: Exception) {
-            callbackObj?.id?.let {
-                bridgeExecutor.postError(
-                    it,
-                    "${ErrorBridgeMessage.ERR_REQ_CUSTOM_PERMISSION} ${e.message}"
-                )
+                }
+            } catch (e: Exception) {
+                if (e.message.toString() == NO_IMPLEMENT_CUSTOM_PERMISSION) {
```

Comment:
The typical way of doing this is to catch different exception types, so it seems a little strange to check the string message instead. Could you make an internal exception type for `CustomPermissionsNotImplementedException`?

Also, should we handle the case here where the Host App throws an exception in `requestCustomPermissions`?

So basically something this:
```
catch (e: CustomPermissionsNotImplementedException) {
  // Display permissions dialog....
} catch (e: Exception) {
  customPermissionBridgeDispatcher.postCustomPermissionError(e.localizedMessage)
}
```

Reply:
I have updated this part using `CustomPermissionsNotImplementedException`.